src/gui/app.cpp

```diff
@@ -1,18 +1,150 @@
 #include "app.h"
 
+#include "arguments.h"
+#include "mainwindow.h"
+#include "printinfo.h"
+#include "version.h"
+
 #include <QDir>
 #include <QFileInfo>
 #include <QFileOpenEvent>
 #include <QSettings>
 #include <QTextStream>
-
-#include "arguments.h"
-#include "mainwindow.h"
-#include "printinfo.h"
-#include "version.h"
+#include <vector>
 
 namespace NeovimQt {
 
+MainWindow* s_lastActiveWindow{ nullptr };
+std::vector<MainWindow*> s_windows;
+int s_exitStatus{ 0 };
+
+struct ConnectorInitArgs
+{
+	enum class Type
+	{
+		Embed,
+		Server,
+		Spawn,
+		Default
+	};
+
+	ConnectorInitArgs(const QCommandLineParser& parser, QStringList nvimArgs)
+		: type{ getConnectorType(parser) }
+		, timeout{ parser.value("timeout").toInt() }
+		, server{ parser.value("server") }
+		, nvim{ parser.value("nvim") }
+		, positionalArgs{ parser.positionalArguments() }
+		, neovimArgs{ std::move(nvimArgs) }
+	{
+	}
+
+	ConnectorInitArgs(
+		Type _type, int _timeout, QString _server, QString _nvim, QStringList nvimArgs)
+		: type{ _type }
+		, timeout{ _timeout }
+		, server{ std::move(_server) }
+		, nvim{ std::move(_nvim) }
+		, positionalArgs{}
+		, neovimArgs{ std::move(nvimArgs) }
+	{
+	}
+
+	const Type type;
+	const int timeout;
+	const QString server;
+	const QString nvim;
+	const QStringList positionalArgs;
+	const QStringList neovimArgs;
+
+	Type getConnectorType(const QCommandLineParser& parser)
+	{
+		if (parser.isSet("server")) {
+			return ConnectorInitArgs::Type::Server;
+		}
+
+		if (parser.isSet("embed")) {
+			return ConnectorInitArgs::Type::Embed;
+		}
+
+		if (parser.isSet("spawn") && !parser.positionalArguments().isEmpty()) {
+			return ConnectorInitArgs::Type::Spawn;
+		}
+
+		return ConnectorInitArgs::Type::Default;
+	}
+};
+
+NeovimConnector* connectToRemoteNeovim(const ConnectorInitArgs& args)
```

Comment:
nitpick:

All new functions and CTORs should be marked as `noexcept`.

Here and elsewhere.